pkg/services/ngalert/notifier/channels/discord.go

```diff
@@ -0,0 +1,124 @@
+package channels
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/grafana/grafana/pkg/bus"
+	"github.com/grafana/grafana/pkg/setting"
+
+	"github.com/grafana/grafana/pkg/components/simplejson"
+
+	gokit_log "github.com/go-kit/kit/log"
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/alerting"
+	old_notifiers "github.com/grafana/grafana/pkg/services/alerting/notifiers"
+	"github.com/prometheus/alertmanager/notify"
+	"github.com/prometheus/alertmanager/template"
+	"github.com/prometheus/alertmanager/types"
+)
+
+type DiscordNotifier struct {
+	old_notifiers.NotifierBase
+	log         log.Logger
+	tmpl        *template.Template
+	Description string
+	Message     string
+	WebhookURL  string
+}
+
+func NewDiscordNotifier(model *models.AlertNotification, t *template.Template) (*DiscordNotifier, error) {
+	if model.Settings == nil {
+		return nil, alerting.ValidationError{Reason: "No Settings Supplied"}
+	}
+
+	discordURL := model.Settings.Get("url").MustString()
+	if discordURL == "" {
+		return nil, alerting.ValidationError{Reason: "Could not find webhook url property in settings"}
+	}
+
+	message := model.Settings.Get("message").MustString()
+
+	return &DiscordNotifier{
+		NotifierBase: old_notifiers.NewNotifierBase(model),
+		Message:      message,
+		WebhookURL:   discordURL,
+		log:          log.New("alerting.notifier.discord"),
+		tmpl:         t,
+	}, nil
+}
+
+func (d DiscordNotifier) Notify(ctx context.Context, as ...*types.Alert) (bool, error) {
+	data := notify.GetTemplateData(ctx, d.tmpl, as, gokit_log.NewNopLogger())
+	alerts := types.Alerts(as...)
+
+	bodyJSON := simplejson.New()
+	bodyJSON.Set("username", "Grafana")
+
+	var tmplErr error
+	tmpl := notify.TmplText(d.tmpl, data, &tmplErr)
+	if d.Message != "" {
+		bodyJSON.Set("content", tmpl(d.Message))
+	}
+	if tmplErr != nil {
+		return false, errors.Wrap(tmplErr, "failed to template discord message")
+	}
+
+	fields := make([]map[string]interface{}, 0)
+	for _, a := range as {
+		fields = append(fields, map[string]interface{}{
+			"name":   a.Labels.String(),
+			"value":  a.Annotations.String(),
+			"inline": true,
+		})
+	}
+
+	footer := map[string]interface{}{
+		"text":     "Grafana v" + setting.BuildVersion,
+		"icon_url": "https://grafana.com/assets/img/fav32.png",
+	}
+
+	embed := simplejson.New()
+	embed.Set("title", getTitleFromTemplateData(data))
```

Comment:
We got rid of this function since it was redundant. Would be a good idea to rebase this PR with `main` once to catch any more changes.
```suggestion
	embed.Set("title", tmpl(`{{ template "default.title" . }}`))
```